client,bridgev2: Add support for MSC4190

onestacked opened the pull request:
Adds [MSC4190](https://github.com/matrix-org/matrix-spec-proposals/pull/4190) support.

tulir reviewed (commented):
The config option goes in `bridgev2/bridgeconfig` + `bridgev2/matrix/mxmain/example-config.yaml` (no need to update the legacy `bridge` module)

`SetAppServiceDeviceID` probably also has to be set in the e2ee client

onestacked commented:
I've now added the config flag in bridgev2 and removed some changes not needed for bridgev2.

I've tested this with element-hq/synapse#17705 and [MSC3861](https://github.com/matrix-org/matrix-spec-proposals/pull/3861).
Requires msc3202_device_masquerading: true to be set in the synapse config.

onestacked commented on bridgev2/bridgeconfig/legacymigrate.go:
```diff
@@ -63,6 +63,7 @@ func doMigrateLegacy(helper up.Helper, python bool) {
 	}
 	helper.Copy(up.Bool, "appservice", "ephemeral_events")
 	helper.Copy(up.Bool, "appservice", "async_transactions")
+	helper.Copy(up.Bool, "appservice", "msc4190")
```
Removed it.

onestacked commented:
@tulir Can you approve the CI?

onestacked commented:
@tulir Friendly ping :)

jkolo commented:
https://github.com/element-hq/synapse/pull/17705 was merged into synapse develop :)
